### PR TITLE
Manager v1.4 adjustments for sanity test support

### PR DESCRIPTION
**Pull request by yarongilor**
Adding the mandatory adjustments in order for scylla-manager sanity test be updated with Manager 1.4 changes (that break backward compatibility)

### CODE REVIEW

**Comment by bentsi** [comment]
what does it mean private?

**Comment by bentsi** [comment]
are we ready to merge this?

**Comment by yarongilor** [comment]
This PR code pass with manager sanity test on 1.4 jenkins job of: 
http://jenkins.cloudius-systems.com:8080/job/mermaid-1.4/job/mermaid-1.4-centos-sanity-test

**Comment by roydahan** [comment]
reviewing again

File: mgmt_cli_test.py [hunk 1]
```diff
@@ -31,7 +31,9 @@ class MgmtCliTest(ClusterTester):
 
     :avocado: enable
     """
-    MANAGER_IDENTITY_FILE = '/tmp/scylla_manager_pem'
+    MANAGER_IDENTITY_FILE_DIR = '/root/.ssh'
```
Comment (bentsi) [inline mgmt_cli_test.py]: You defined it in `mgmt.py`, why did you copy-paste it here?

File: sdcm/cluster.py [hunk 2]
```diff
@@ -1334,6 +1335,7 @@ def install_mgmt(self, scylla_repo, scylla_mgmt_repo):
 
         if self.is_rhel_like():
             self.remoter.run('sudo yum install -y epel-release', retry=3)
+            self.remoter.run('sudo yum install python36-PyYAML -y', retry=3)
```
[the remark this answers is not in the document]
Comment (yarongilor) [reply sdcm/cluster.py]: hi @amoskong , can you please confirm this WA isn't needed anymore?

**Comment by yarongilor** [comment]
> one change needed.
> 
> another question, @yarongilor does it support the upgrade procedure?
> After having 1.3 installed and configured with the old method, would it start normally after upgrade?

currently any SCT Manager branch supports only "future" upgrades.
that means, for example, that 1.4 doesn't have backwards compatibility for 1.3.1 and cannot test upgrade of 1.3.1 -> 1.4. 
so currently, in order to test 1.3.1 -> 1.4 upgrade, the jenkins job uses SCT 1.3.1 branch.

we can add a task of backward compatibility to SCT as well.

File: mgmt_cli_test.py (hunk 1 above)
Comment (yarongilor) [reply mgmt_cli_test.py]: removed duplicate.

File: sdcm/cluster.py (hunk 2 above)
[the remark this answers is not in the document]
Comment (amoskong) [reply sdcm/cluster.py]: We don't need it anymore.
Patch for Scylla 3.0 / 2019.1 / 2.3 had been merged, patch for 2018.1 is in queue.

**Comment by roydahan** [comment]
@bentsi let's merge and backport.

**Pull request merged**